### ISSUE TITLE
Add automatic resizing

### ISSUE THREAD

**Issue by PK-cod3ch3mist** [PK-cod3ch3mist/ASCII-Media-Player#17]
Use the curses module to detect change in terminal size and automatically resize the video render processes to adapt to it.
You can take inspiration and check the size transformation functions in the `get_pixel_matrix` function.

**Comment by Tom712** [comment]
Hello, I would like to work on this issue could you assign it to me please?

**Comment by PK-cod3ch3mist** [comment]
Hi @Tom712, sure. Do you have an approach in mind?

**Comment by Tom712** [comment]
I am thinking about using opencv library to get the video resolution and then resize the terminal using that. Need to experiment with it a little past that.

**Comment by PK-cod3ch3mist** [comment]
Hi @Tom712, I think that you possibly misunderstood the issue. There already is a function that changes the resolution of the video based on the current terminal size and font. This is the `get_pixel_matrix` function.

I do not intend to do this the other way round. Because it should be the user's choice to resize the terminal.

However, what we do need is if te user resizes the terminal mid render, the output should not overflow the terminal boundaries and the video resolution should again be changed accordingly.

Basically check if terminal is resized, if yes, call `get_pixel_matrix` and resize media and caption windows.

**Comment by Tom712** [comment]
Alright, I'll admit I misunderstood the issue however, I would still like to work on it.

**Comment by Tom712** [comment]
@PK-cod3ch3mist  Hello,  will you assign me the issue?

**Comment by PK-cod3ch3mist** [comment]
Yes okay @Tom712.

**Comment by PK-cod3ch3mist** [comment]
Hi @Tom712, any updates on the automatic resize?

**Comment by Tom712** [comment]
Sorry not yet. I had to focus on my school assignment but i'll work on this week.

**Comment by Tom712** [comment]
@PK-cod3ch3mist Yes I am working on it, I am just on a holiday with family right now, sorry for the late reply.

**Comment by Tom712** [comment]
@PK-cod3ch3mist Hello I have finished the automatic resizing and created a pull request.
I'll admit it took me too long do this but hopefully its satisfactory.
I already mentioned this in the description of the pull request but i'll say it here again I had to change the import of cv2 because it wouldnt work otherwise on a fresh Virtual Machine running Ubuntu.
